### PR TITLE
docs(animations): add playground for preference based

### PR DESCRIPTION
**Pull request by thetaPC**
This is an alternative implementation of https://github.com/ionic-team/ionic-docs/pull/1712 which replaces the "Preference-Based Animations" samples with a playground. The playground contains Vanilla JS, Angular, React, and Vue code samples, and it also links out to StackBlitz.

[Demo URL](https://ionic-docs-git-fw-3376-ionic1.vercel.app/docs/utilities/animations#preference-based-animations)

---
Change the theme through your OS's system settings to view the differences or change the theme via the site.

### CODE REVIEW

File: static/usage/v7/animations/preference-based/vue.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+```html
+<template>
+  <ion-card ref="cardEl">
+    <ion-card-content>Card</ion-card-content>
+  </ion-card>
+
+  <ion-button id="play" @click="play()">Play</ion-button>
+  <ion-button id="pause" @click="pause()">Pause</ion-button>
+  <ion-button id="stop" @click="stop()">Stop</ion-button>
```
Comment (brandyscarney) [inline static/usage/v7/animations/preference-based/vue.md:9]:
```suggestion
  <ion-button @click="play()">Play</ion-button>
  <ion-button @click="pause()">Pause</ion-button>
  <ion-button @click="stop()">Stop</ion-button>
```

File: static/usage/v7/animations/preference-based/react/main_tsx.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+      <IonButton id="play" onClick={play}>
+        Play
+      </IonButton>
+      <IonButton id="pause" onClick={pause}>
+        Pause
+      </IonButton>
+      <IonButton id="stop" onClick={stop}>
+        Stop
+      </IonButton>
```
Comment (brandyscarney) [inline static/usage/v7/animations/preference-based/react/main_tsx.md:48]:
```suggestion
      <IonButton onClick={play}>
        Play
      </IonButton>
      <IonButton onClick={pause}>
        Pause
      </IonButton>
      <IonButton onClick={stop}>
        Stop
      </IonButton>
```

File: static/usage/v7/animations/preference-based/javascript.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+```html
+<ion-card id="card">
+  <ion-card-content>Card</ion-card-content>
+</ion-card>
+
+<ion-button id="play" onclick="animation.play()">Play</ion-button>
+<ion-button id="pause" onclick="animation.pause()">Pause</ion-button>
+<ion-button id="stop" onclick="animation.stop()">Stop</ion-button>
```
Comment (brandyscarney) [inline static/usage/v7/animations/preference-based/javascript.md:8]:
```suggestion
<ion-button onclick="animation.play()">Play</ion-button>
<ion-button onclick="animation.pause()">Pause</ion-button>
<ion-button onclick="animation.stop()">Stop</ion-button>
```

File: static/usage/v7/animations/preference-based/angular/example_component_html.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+```html
+<ion-card #card>
+  <ion-card-content>Card</ion-card-content>
+</ion-card>
+
+<ion-button id="play" (click)="play()">Play</ion-button>
+<ion-button id="pause" (click)="pause()">Pause</ion-button>
+<ion-button id="stop" (click)="stop()">Stop</ion-button>
```
Comment (brandyscarney) [inline static/usage/v7/animations/preference-based/angular/example_component_html.md:8]:
```suggestion
<ion-button (click)="play()">Play</ion-button>
<ion-button (click)="pause()">Pause</ion-button>
<ion-button (click)="stop()">Stop</ion-button>
```

**Review by brandyscarney — COMMENTED** [review]
The StackBlitz demo buttons don't need `id`s because they call the click directly

**Review by amandaejohnston — COMMENTED** [review]
Now we sort of have the opposite problem -- if I have my device set to dark mode, but switch to light mode in Docusaurus, I get the green background, since the `prefers-color-scheme` kicks in but the `body.dark` doesn't.

My immediate thought is to do `body:not(.dark) ion-card { --background: red }` but then we're coupling it entirely to the Docusaurus setting, not the device 😆 I'm on the fence about whether we should have the Docusaurus fix at all. Maybe there's an easy way to fix all this that I'm not seeing at the moment?

**Comment by brandyscarney** [comment]
Yeah I am not sure either. I think it's the fact that the body of the app doesn't change with the prefers color scheme setting, only the card does. However, the sites dark mode toggle is tied to the system settings so the body will always change with that.

I think we should match the way the dark mode toggle prioritizes it just for the demo, the StackBlitz code would still only use the prefers-color-scheme.

**Comment by thetaPC** [comment]
I changed it to only rely on Docusaurus since it can already determine when the theme is changed through the site or the OS settings. It doesn't match the StackBlitz code but it should be fine since it still portrays the end result.

**Review by amandaejohnston — APPROVED** [review]
Good point, this works great 👍

**Pull request merged**